pkg/daemon/ceph/osd/daemon.go

```diff
@@ -250,10 +250,10 @@ func getDataDirs(context *clusterd.Context, kv *k8sutil.ConfigMapKVStore, desire
 		dirList = strings.Split(desiredDirs, ",")
 	}
 
-	if len(dirList) == 0 && !devicesSpecified {
-		// user has not specified any dirs or any devices, give them the default dir at least
-		dirList = append(dirList, context.ConfigDir)
-	}
+	// user has not specified any dirs or any devices, legacy behavior was to give them the default
```

Comment:
the start of this comment confused me for a second because it's phrased as if we have already reached this condition at this point of the code flow.  maybe just add "when" or "if" to make it more explicit.  e.g.:
```
// when the user has not specified any dirs or any devices, ...
```